### PR TITLE
added spikegadgets data interface with new source_schema setting method

### PR DESCRIPTION
**Pull request by CodyCBakerPhD**
## Motivation
SpikeGadgetsRawIO has been merged to neo master ([PR #954](https://github.com/NeuralEnsemble/python-neo/pull/954)), and SpikeGadgetsRecordingExtractor has been merged to spikeextractors master ([PR #592](https://github.com/SpikeInterface/spikeextractors/pull/592)).

This is the next step of building the `SpikeGadgetsDataInterface`. Pretty simple, doesn't require a lot of extra setting in terms of metadata, aside from the optional probe file.

## How to test the behavior?
Existing schema tests should cover the new interface since it was added to the `interface_list`.

Other than that, a big to-do for this and all the other data interfaces is to add GIN and other running tests to ensure they function as expected.

## Checklist

- [X] Have you checked our [Contributing](https://github.com/catalystneuro/nwb-conversion-tools/blob/master/docs/contribute.rst) document?
- [X] Have you ensured the PR description clearly describes the problem and solutions?
- [X] Is your contribution compliant with our coding style? Read about [PEP8](https://www.python.org/dev/peps/pep-0008/) and [black](https://black.readthedocs.io/en/stable/the_black_code_style.html).
- [X] Have you checked to ensure that there aren't other open or previously closed [Pull Requests](https://github.com/catalystneuro/nwb-conversion-tools/pulls) for the same change?

### CODE REVIEW

**Comment by CodyCBakerPhD** [comment]
@luiztauffer We'll certainly get a better look once the other CI workflow PR is merged, but right now the AxonaPositionInterface is causing new issue with some annotations, any clue why?

Also interesting to note that when I run the tests locally on Windows, I see similar issues with the Hdf5 and Tiff Imaging interfaces...

**Review by bendichter — CHANGES_REQUESTED** [review]
Looks good. Are we able to create tests for this as well?

File: nwb_conversion_tools/datainterfaces/ecephys/spikegadgets/spikegadgetsdatainterface.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    def get_source_schema(cls):
+        source_schema = get_schema_from_method_signature(cls.__init__)
+        source_schema["properties"]["filename"].update(format="file", description="Path to SpikeGadgets (.rec) file.")
+        source_schema["properties"]["probe_file_path"].update(
+            format="file",
+            description="Optional path to a probe (.prb) file describing electrode features."
+        )
+        return source_schema
+
+    def __init__(self, filename: PathType, probe_file_path: OptionalPathType):
```
Comment (bendichter) [inline nwb_conversion_tools/datainterfaces/ecephys/spikegadgets/spikegadgetsdatainterface.py:29]:
```suggestion
    def __init__(self, filename: PathType, probe_file_path: OptionalPathType = None):
```

File: nwb_conversion_tools/datainterfaces/ecephys/spikegadgets/spikegadgetsdatainterface.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        super().__init__(filename=filename)
+        self.recording_extractor = load_probe_file(recording=self.recording_extractor, probe_file=probe_file_path)
```
[the remark this answers is not in the document]
Comment (CodyCBakerPhD) [reply nwb_conversion_tools/datainterfaces/ecephys/spikegadgets/spikegadgetsdatainterface.py:31]: Whoops... adding now

**Comment by CodyCBakerPhD** [comment]
@bendichter When I add the GIN tests in their entirety, the writing of actual spikegadgets data (a part of which is using data from the Brody lab itself) will be among the formats covered.

I'll try to get that up later this week.

**Comment by luiztauffer** [comment]
@CodyCBakerPhD 
I removed an unnecessary argument from the `AxonaPositionDataInterface.run_conversion()` method and this solved the previous error.
Right now we have the CI passing for the development env, but not for the production env.

**Review by luiztauffer — APPROVED** [review]

**Comment by CodyCBakerPhD** [comment]
@luiztauffer Cool, that's to be expected pending the next neo/se releases. The production testing may have also revealed some new version dependence issues we can work through in a subsequent PR. Merging for now so we can get this dev fix in at least

**Pull request merged**